src/VirtualDevice.ts

```diff
@@ -146,14 +151,40 @@ export class VirtualDevice {
 
     private handleBatchResponse(data: string): IVirtualDeviceResult[] {
         const json = JSON.parse(data);
+        for (const result of json.results) {
+            this.applyHomophones(result);
+        }
         return json.results;
     }
+
+    private applyHomophones(result: IVirtualDeviceResult) {
+        if (!result.debug) {
+            result.debug = {};
+        }
+
+        if (!result.transcript) {
+            return;
+        }
+
+        const keys = Object.keys(this.homophones);
+        result.debug.rawTranscript = result.transcript;
+
+        for (const key of keys) {
+            const homophones = this.homophones[key];
+            for (const homophone of homophones) {
+                // Replace each of the homophones
+                result.transcript = result.transcript.split(homophone).join(key);
```

Comment:
I think we should use some sort of regular expression instead of a simple split in order to avoid unwanted replacements. Example: having "to" as the key with "two" as an homophone, I think this code could change words like "together", "tomorrow", etc. into "twogether", "twomorrow" and so on.

Also, looping the homophones of a key to do the replacements makes me think that only the first one is going to be applied. Is that correct?

Reply:
It applies them to any homophone, not just the first one, you can see that in the tests.